### PR TITLE
Add imagePullPolicy flag to istioctl and kube platform

### PR DESCRIPTION
**Pull request by cmluciano**
Signed-off-by: Christopher M. Luciano <cmluciano@us.ibm.com>

Fixes: #989

### CODE REVIEW

**Comment by cmluciano** [comment]
Notes for reviewers:

- Does a string value with a default make sense or would you rather have a bool option like --imagePullAlways?
- Does the default of Always make sense given proxy is using IfNotPresent?
- Based on feedback I will update the e2e tests to use IfNotPresent?

File: platform/kube/inject/inject.go [hunk 1]
```diff
@@ -143,13 +144,20 @@ func injectIntoPodTemplateSpec(p *Params, t *v1.PodTemplateSpec) error {
 		initArgs = append(initArgs, "-i", p.IncludeIPRanges)
 	}
 
+	var pullPolicy v1.PullPolicy
```
Comment (ayj) [inline platform/kube/inject/inject.go]: This should have a reasonable default or return an error for unknown values.

File: cmd/istioctl/inject.go [hunk 2]
```diff
@@ -159,6 +161,8 @@ func init() {
 	injectCmd.PersistentFlags().BoolVar(&enableCoreDump, "coreDump",
 		true, "Enable/Disable core dumps in injected Envoy sidecar (--coreDump=true affects "+
 			"all pods in a node and should only be used the cluster admin)")
+	injectCmd.PersistentFlags().StringVar(&imagePullPolicy, "imagePullPolicy", "Always",
+		"Sets the container image pull policy to Always or PullIfNotPresent")
```
Comment (ayj) [inline cmd/istioctl/inject.go]: re: string vs. bool, the description suggests either/or behavior (e.g. bool) but its really a string value which is confusing. Maybe just update the description to include the list of valid options [Always,Never,IfNotPresent]?

File: platform/kube/inject/inject.go [hunk 3]
```diff
@@ -143,13 +144,20 @@ func injectIntoPodTemplateSpec(p *Params, t *v1.PodTemplateSpec) error {
...
+	if p.ImagePullPolicy == "Always" {
```
Comment (ijsnellf) [inline platform/kube/inject/inject.go]: style nit: a `switch` would be more concise

Comment (cmluciano) [inline platform/kube/inject/inject.go]: ack

**Pull request merged**